### PR TITLE
Added controllers and mappers for CinemaHall, Movie and MovieSession

### PR DESCRIPTION
**Pull request by vicire**

### CODE REVIEW

File: src/main/java/cinema/controller/CinemaHallController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    @PostMapping
+    public void create(@RequestBody CinemaHallRequestDto cinemaHallDto) {
+        CinemaHall cinemaHall = cinemaHallMapper.toCinemaHallEntity(cinemaHallDto);
+        cinemaHallService.add(cinemaHall);
+    }
+
+    @GetMapping
+    public List<CinemaHallResponseDto> getAll() {
+        List<CinemaHall> allCinemaHalls = cinemaHallService.getAll();
+        return allCinemaHalls.stream()
```
Comment (Max-Pochepets) [inline src/main/java/cinema/controller/CinemaHallController.java]: You can inline this

File: src/main/java/cinema/controller/MovieSessionController.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    @PutMapping
+    public void update(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.update(movieSessionMapper.toMovieSessionEntity(movieSessionRequestDto));
+    }
+
+    @DeleteMapping
+    public void delete(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        movieSessionService.delete(movieSessionMapper.toMovieSessionEntity(movieSessionRequestDto));
+    }
```
Comment (Max-Pochepets) [inline src/main/java/cinema/controller/MovieSessionController.java]: Check [this](https://mateacademy.slack.com/archives/G01EHB8CWAE/p1613330083125300) please

File: src/main/java/cinema/model/dto/request/MovieSessionRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package cinema.model.dto.request;
+
+public class MovieSessionRequestDto {
+    private Long movieSessionId;
```
Comment (Max-Pochepets) [inline src/main/java/cinema/model/dto/request/MovieSessionRequestDto.java]: We don't need this here, you'll see why

**Review by Max-Pochepets — CHANGES_REQUESTED** [review]

File: src/main/java/cinema/controller/CinemaHallController.java (hunk 1 above)
Comment (vicire) [reply src/main/java/cinema/controller/CinemaHallController.java]: Fixed

File: src/main/java/cinema/controller/MovieSessionController.java (hunk 2 above)
Comment (vicire) [reply src/main/java/cinema/controller/MovieSessionController.java]: Fixed

File: src/main/java/cinema/model/dto/request/MovieSessionRequestDto.java (hunk 3 above)
Comment (vicire) [reply src/main/java/cinema/model/dto/request/MovieSessionRequestDto.java]: Fixed

File: src/main/java/cinema/controller/MovieSessionController.java [hunk 4]
```diff
@@ -47,13 +48,16 @@ public void create(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
...
+        MovieSession movieSession = movieSessionMapper.toMovieSessionEntity(movieSessionRequestDto);
+        movieSession.setId(id);
+        movieSessionService.update(movieSession);
     }
 
-    @DeleteMapping
-    public void delete(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
-        movieSessionService.delete(movieSessionMapper.toMovieSessionEntity(movieSessionRequestDto));
+    @DeleteMapping("/{id}")
+    public void delete(@PathVariable("id") Long id) {
```
Comment (Max-Pochepets) [inline src/main/java/cinema/controller/MovieSessionController.java]:
```suggestion
    public void delete(@PathVariable Long id) {
```
If names match - no need to specify Path Variable name

**Review by Max-Pochepets — APPROVED** [review]

File: src/main/java/cinema/controller/MovieSessionController.java (hunk 4 above)
Comment (vicire) [reply src/main/java/cinema/controller/MovieSessionController.java]: Fixed

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**